File: chroma-league/src/test/java/net/booone/chromaleague/rest/http/handlers/EventsResponseHandlerTest.java
package net.booone.chromaleague.rest.http.handlers;

import net.booone.chromaleague.rest.eventdata.Event;
import net.booone.chromaleague.rest.http.LeagueHttpClientMock;
import org.apache.hc.core5.http.ClassicHttpResponse;
import org.apache.hc.core5.http.HttpException;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.util.List;
import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class EventsResponseHandlerTest {

    @Test
    void testEventParsing() throws IOException, URISyntaxException, HttpException {
        //given
        ClassicHttpResponse testResponseFromJSON = LeagueHttpClientMock.createTestResponseFromJSON("json/standardevent.json");

        //when
        Optional<List<Event>> events = new EventsResponseHandler().handleResponse(testResponseFromJSON);

        //then
        assertTrue(events.isPresent());
        verifyStandardEvent(events.get());
    }

    public static void verifyStandardEvent(List<Event> events) {
        assertEquals(3, events.size());

        final Event event0 = events.get(0);
        assertEquals(0, event0.EventID());
        assertEquals("GameStart", event0.EventName());
        assertEquals(0.0563616007566452, event0.EventTime());

        final Event event1 = events.get(1);
        assertEquals(1, event1.EventID());
        assertEquals("MinionsSpawning", event1.EventName());
        assertEquals(5.06690979003906, event1.EventTime());

        final Event event2 = events.get(2);
        assertEquals(2, event2.EventID());
        assertEquals("ChampionKill", event2.EventName());
        assertEquals(7.1595916748047, event2.EventTime());
    }
}

File: test/rest/http/handlers/EventsResponseHandler.test.ts
import {describe, expect, test} from 'vitest';
import type {Event} from '../../../../src/rest/eventdata/Event.js';
import type {Events} from '../../../../src/rest/eventdata/Events.js';

describe('EventsResponseHandler', () => {
    test('testEventParsing', () => {
        //given
        const json: Events = {
            Events: [
                { EventID: 0, EventName: 'GameStart', EventTime: 0.0563616007566452 },
                { EventID: 1, EventName: 'MinionsSpawning', EventTime: 5.06690979003906 },
                { EventID: 2, EventName: 'ChampionKill', EventTime: 7.1595916748047, KillerName: 'BooonE', VictimName: 'Warwick Bot', Assisters: [] },
            ],
        };

        //when
        const events = json.Events;

        //then
        expect(events).toBeDefined();
        verifyStandardEvent(events);
    });
});

export function verifyStandardEvent(events: Event[]): void {
    expect(events).toHaveLength(3);

    const event0 = events[0];
    expect(event0.EventID).toBe(0);
    expect(event0.EventName).toBe('GameStart');
    expect(event0.EventTime).toBe(0.0563616007566452);

    const event1 = events[1];
    expect(event1.EventID).toBe(1);
    expect(event1.EventName).toBe('MinionsSpawning');
    expect(event1.EventTime).toBe(5.06690979003906);

    const event2 = events[2];
    expect(event2.EventID).toBe(2);
    expect(event2.EventName).toBe('ChampionKill');
    expect(event2.EventTime).toBe(7.1595916748047);
}
